tips/TIP-0023/tip-0023.md

```diff
@@ -0,0 +1,47 @@
+---
+tip: 23
+title: Tagged Data Payload
+description: Payload for arbitrary data
+author: Wolfgang Welz (@Wollac) <wolfgang.welz@iota.org>
+discussions-to: https://github.com/iotaledger/tips/pull/54
+status: Draft
+type: Standards
+layer: Core
+created: 2022-01-24
+---
+
+# Abstract
+
+The payload concept offers a very flexible way to combine and encapsulate information in the IOTA protocol. This document proposes a basic payload type that allows the addition of arbitrary data.
+
+# Motivation
+
+The most flexible way to extend an existing object is by the addition of arbitrary data. This payload provides a way to do just that. An optional tag can be used to categorize the data.
+
+# Specification
+
+## Serialized Layout
+
+The following table describes the serialization of a _Tagged Data Payload_ following the notation from [TIP-21](../TIP-0021/tip-0021.md):
+
+| Name         | Type                  | Description                                          |
+| ------------ | --------------------- | ---------------------------------------------------- |
+| Payload Type | uint32                | Set to *value 5* to denote an _Tagged Data Payload_. |
+| Tag Length   | uint8                 | The length of the following tag field in bytes.      |
+| Tag          | ByteArray[Tag Length] | The tag of the data.                                 |
+| Data         | ByteArray             | Binary data.                                         |
```

Comment:
how do we know the length of the arbitrary data, shouldn't have a `Data Length` field? 
Is max data length depend on parent number and tag length?

Reply:
according to TIP-18, all ByteArray paired with a length field in structures. for instance `State Metadata` has `State Metadata Length` in the [Alias output](https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#alias-output). serialization of State Metadata would be `uint16 + uint32 + data bytes` which is not correct. 
the definition of ByteArray in TIP-21 is not true since the length limitation of ByteArray is vary depends on the object.